contracts/smart-contract-wallet/SmartAccount.sol

```diff
@@ -206,19 +206,29 @@ contract SmartAccount is
     }
 
     /**
-     * @dev returns a value from the nonces 2d mapping
+     * @dev for batchId = 0 returns nonce(), that uses EntryPoint's nonce if EntryPoint supports it
+     *      for other batchIds returns a value from the nonces 2d mapping
      * @param batchId : the key of the user's batch being queried
      * @return nonce : the number of transactions made within said batch
      */
-    function getNonce(uint256 batchId) public view returns (uint256) {
+    function getNonce(uint256 batchId) public view virtual returns (uint256) {
+        if (batchId == 0) 
+            return nonce();
         return nonces[batchId];
     }
 
     /**
-     * @dev Standard interface for 1d nonces. Use it for Account Abstraction flow.
+     * @dev Returns a nonce from EntryPoint if EntryPoint supports it. Otherwise returns internal AA-flow nonce.
      */
     function nonce() public view virtual override returns (uint256) {
```

Comment:
at this point I am also thinking why limit keySpace 0
abi.encodeWithSignature("getNonce(address,uint192)", address(this), 0)
if it allows for bundlers to send parallel transactions

Reply:
that would require getNonce() method for AA and getForwardNonce() for nonERC4337 transactions (name inspired from other PR https://github.com/bcnmy/scw-contracts/pull/76 )

Reply:
yes, EP's 2d nonces would require introducing separate method for forward flow.
It can be actually a good idea since we've discussed on yesterday's call that AA flow should be the default flow.